version: "3.3"

networks:
  reverse-proxy:
    external:
      name: ${REVERSE_PROXY_NETWORK}
services:

  portainer:
    container_name: portainer
    image: portainer/portainer
    restart: always
    command: --no-auth -H unix:///var/run/docker.sock
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    networks:
      - ${REVERSE_PROXY_NETWORK}
    labels:
      - "traefik.enable=true"
      - "traefik.port=9000"
      - "traefik.frontend.rule=Host:portainer.${BASE_DOMAIN}"

  traefik:
    container_name: traefik
    image: traefik:tetedemoine
    restart: always
    command: |
      -c /dev/null \
      --web \
      --docker \
      --docker.domain=${BASE_DOMAIN} \
      --docker.exposedbydefault=false \
      --logLevel=ERROR \
      --entrypoints="Name:http Address::80 Redirect.EntryPoint:https" \
      --entryPoints="Name:https Address::443 TLS:/tools/certs/docker.devhost.crt,/tools/certs/docker.devhost.key" \
      --defaultentrypoints="http,https"
    volumes:
      - $PWD/ssl/certificats:/tools/certs
      - /var/run/docker.sock:/var/run/docker.sock
    networks:
      - ${REVERSE_PROXY_NETWORK}
    ports:
      - 80:80
      - 443:443
    labels:
        traefik.enable: "true"
        traefik.port: "8080"
        traefik.backend: "traefik"
        traefik.frontend.rule: "Host:traefik.${BASE_DOMAIN}"
